src/elements/content-sidebar/activity-feed/app-activity/AppActivity.tsx

```diff
@@ -0,0 +1,179 @@
+import type { User, BoxItemPermission } from '../../../../common/types/core';
+import './AppActivity.scss';
+
+interface AppActivityProps {
+    activity_template: ActivityTemplateItem;
+    app: AppItem;
+    created_at: string;
+    created_by: User;
+    currentUser?: User;
+    error?: ActionItemError;
+    id: string;
+    intl: IntlShape;
+    isPending?: boolean;
+    onDelete: ({ id, permissions }: { id: string; permissions?: BoxItemPermission }) => void;
+    permissions?: BoxItemPermission;
+    rendered_text: string;
+}
+
+interface State {
+    isConfirmingDelete: boolean;
+}
+
+function mapActivityNodes(node: HTMLLinkElement): React.ReactNode {
+    const { dataset = {}, href = '#', tagName, textContent } = node;
+
+    switch (tagName) {
+        case 'A':
+            return (
+                <Link
+                    href={href}
+                    data-resin-target={dataset.resinTarget}
+                    data-resin-action={dataset.resinAction}
+                    key={`app_actvity_link_${href}`}
+                    rel="roreferrer noopener"
+                    className="bcs-AppActivity-link"
+                    target="_blank"
+                >
+                    {textContent}
+                </Link>
+            );
+        default:
+            return textContent;
+    }
+}
+
+class AppActivity extends React.PureComponent<AppActivityProps, State> {
+    static defaultProps = {
+        onDelete: noop,
+        permissions: {},
+    };
+
+    parser = new DOMParser();
+
+    state: State = {
+        isConfirmingDelete: false,
+    };
+
+    handleDeleteCancel = (): void => {
+        this.setState({ isConfirmingDelete: false });
+    };
+
+    handleDeleteClick = (): void => {
+        this.setState({ isConfirmingDelete: true });
+    };
+
+    handleDeleteConfirm = (): void => {
+        const { id, onDelete, permissions } = this.props;
+
+        onDelete({ id, permissions });
+    };
+
+    parseActivity = (): Array<HTMLLinkElement> => {
+        const { rendered_text: renderedText } = this.props;
+        const doc: Document = this.parser.parseFromString(renderedText, 'text/html');
+        if (!doc) {
+            return [];
+        }
+
+        const childNodes = getProp(doc, 'body.childNodes', []);
+        return Array.from(childNodes);
+    };
+
+    render() {
+        const {
+            activity_template: { id: templateId },
+            app: { name, icon_url },
+            created_at: createdAt,
+            created_by: createdBy,
+            currentUser,
+            error,
+            intl,
+            isPending,
+            permissions,
+        } = this.props;
+
+        const canDelete = getProp(permissions, 'can_delete', false) || (currentUser && currentUser.id === createdBy.id);
+        const createdAtTimestamp = new Date(createdAt).getTime();
+        const isMenuVisible = canDelete && !isPending;
+        const { isConfirmingDelete } = this.state;
+
+        return (
```

Comment:
Please delete the empty lines in this JSX return statement. Only delete the empty lines within this return, do not modify other parts of the file.